.github/workflows/pre-commit-ci.yml

```diff
@@ -0,0 +1,19 @@
+name: pre-commit-ci
+on:
+  pull_request:
+    branches: [master]
+jobs:
+  pre-commit-ci:
+    name: Run pre-commit checks on changed files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: git fetch --depth=1 origin +${{github.base_ref}}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - uses: pre-commit/action@v2.0.0
```

Comment:
The pre-commit action version is separate from the pre-commit version which is constrained by our testing on Python 3.7. So I was able to make the requested change.

Reply:
does deprecating 3.7 in #287 remove this constraint?

Reply:
@gwaybio There is no constraint relevant to this pull request. The version of the precommit github action (here) is not related to the precommit package version (which is constrained by Python 3.7).

Reply:
got it! we should keep the precommit version here consistent with requirements-dev.txt (https://github.com/cytomining/pycytominer/issues/273#issuecomment-1585987541)

Reply:
@gwaybio I think there might still be some confusion here. Both the Pre-commit github action (currently at 3.0.0) and the pre-commit python package (3.2.2) share a major version number but they are completely unrelated versions. The pre-commit github action (as show [here](https://github.com/pre-commit/action/blob/main/action.yml)) actually installs and uses the latest version of pre-commit available from pip rather than specifically installing 3.0.0.

So perhaps it would be better to set the pre-commit version number in requirements-dev.txt to `>=3.2.2 ` instead of `==3.2.2` so both the github action precommit and the dev environment precommit stay better in sync. Regardless, because pre-commit is essentially just a wrapper for launching specific tools that have their own versioning, I think we are unlikely to run into compatibility issues with this loose versioning scheme.

Reply:
> Both the Pre-commit github action (currently at 3.0.0) and the pre-commit python package (3.2.2) share a major version number but they are completely unrelated versions.

Gotcha! Thanks for this explanation.

> So perhaps it would be better to set the pre-commit version number in requirements-dev.txt to >=3.2.2 instead of ==3.2.2 so both the github action precommit and the dev environment precommit stay better in sync.

I agree. Will do!

Reply:
Done in 61681979f86f5650e8036c1fdf75806cbd5f19a5